Fix more issues

pharaox opened the pull request:
Fix more issues I encountered while playing beta2 and integrating with other mods, reported either by ck3-tiger or the game itself. Unlike the issues fixed in #221 , these could be just unfinished stuff, so I open this PR more to raise awareness about them.

* Missing Hangzhou events, modifiers, opinion modifiers, etc.
* Missing Sicily FP game concepts (these tend to clutter `error.log` quite a lot), since Sicily is an old FP I am not sure if you are aware of these and plan to fix them.
* Missing culture namelist names, these also tend to show up in `error.log` quite a lot.
* `RICE_SEA_harvest_rare_tropical_goods_interaction_effect` cost set to `gold` without any min or max (I thought this is probably not intended, as it could cost 1 or 1000)

cybrxkhan commented:
Yeah, some of these are unfinished stuff. But a few of the ones you had are minor boo-boos I should take care of, so I'll change those on my local build, no need to worry about those on your end. I'll leave this thread open however in case you have any other thoughts or need to mention anything else.

For the Sicily game concepts, there was actually a vanilla bug where the more game concepts you had, due to some hash collision thing, there was a chance the game would CTD while loading, roughly when the main menu kicks in, but before you load into the menu. It had been an issue plaguing some of the TCs for at least a year now, and I got hit with it while updating RICE to AUH (which is why those game concepts exist in the RICE versions before AUH with no problem). So I took out those game concepts temporarily since apparently they were the ones with hash collisions in this version of RICE. The most recent AUH patch should have fixed the issue, so I plan to re-add those game concepts.

Thanks again for the help :)

pharaox commented:
Thanks very much for explaining! I wasn't aware of the CTD and I am playing with those game concepts added, no crashes so far.

pharaox committed:
Fix various issues

pharaox commented on RICE/common/scripted_effects/RICE_aslian_effects.txt:
```diff
@@ -476,7 +476,7 @@ RICE_SEA_harvest_rare_tropical_goods_interaction_effect = {
 	}
 	scope:actor = {
-		remove_short_term_gold = gold
+		remove_short_term_gold = 25 # gold
```
This probably shouldn't cost **all your gold**?

pharaox commented on RICE/common/scripted_effects/RICE_nubia_effects.txt:
```diff
@@ -4356,7 +4356,10 @@ RICE_christian_commission_donor_portrait_effect = {
 
 RICE_christian_commission_donor_portrait_base_effect = {
-	add_legitimacy = RICE_donor_portrait_base_legitimacy_gain
+	if = {
+		limit = { has_legitimacy = yes }
+		add_legitimacy = RICE_donor_portrait_base_legitimacy_gain
+	}
```
Avoid errors for rulers without legitimacy (apparently they can take the decision as well)

pharaox commented on RICE/common/scripted_effects/RICE_sardinia_effects.txt:
```diff
@@ -307,7 +307,10 @@ RICE_setup_afro_roman_counties_effect = {
 					OR = {
 						culture = culture:african_roman
-						culture = culture:afro_roman
+						AND = {
+							has_global_variable = RICE_game_started_on_TFE_start_date
+							culture = culture:afro_roman
+						}
```
Avoid errors for missing cultures if not using TFE

pharaox commented on RICE/common/scripted_effects/RICE_siwa_effects.txt:
```diff
@@ -113,13 +113,11 @@ make_siwi_culture_effect = {
 		limit = {
 			any_vassal = {
 				is_ai = yes
-				culture = root.culture
```
`root` doesn't exist here, and I couldn't figure out what other scope you could have meant, so I just removed these 2 lines

pharaox commented on RICE/localization/english/rice_religions_l_english.yml:
```diff
@@ -902,7 +902,8 @@
  game_concept_yazata:0 "Yazata"
  game_concept_yazatas:0 "Yazatas"
- game_concept_yazatas_desc:0 "The Yazatas are those who are worthy of worship or veneration in $zoroastrianism_religion$. To some, they are distinct deities; to others, they are emanations of the supreme god.\n\n$zoroastrianism_religion_adj$ [faiths|E] can have different [doctrines|E] representing the set of Yazatas that have been important to them due to reasons of history, culture, geography, a ...
+ game_concept_yazata_desc:0 "The Yazatas are those who are worthy of worship or veneration in $zoroastrianism_religion$. To some, they are distinct deities; to others, they are emanations of the supreme god.\n\n$zoroastrianism_religion_adj$ [faiths|E] can have different [doctrines|E] representing the set of Yazatas that have been important to them due to reasons of history, culture, geography, an ...
```
Not having this generates an error for a missing concept localization.

pharaox commented:
I rebased this PR on master, removed all stuff that was no longer relevant, and added a few more fixes for errors I observed while playing. I think it could be reviewed and merged now.

cybrxkhan commented:
Thanks for the help! 🫡
